### PR TITLE
allow optional usage of use_caffe_datum in resnet50 trainer

### PR DESCRIPTION
**Pull request by neobarney**
set use_caffe_datum arg in AddImageInput to False as default value, and added use_caffe_datum as new optional argument in argparse

### CODE REVIEW

**Comment by neobarney** [comment]
Ok I signed up to the CLA

File: caffe2/python/examples/resnet50_trainer.py [hunk 1]
```diff
@@ -446,6 +448,8 @@ def main():
                         help="Save the trained model to a given name")
     parser.add_argument("--load_model_path", type=str, default=None,
                         help="Load previously saved model to continue training")
+    parser.add_argument("--use_caffe_datum", type=bool, default=False,
```
Comment (akyrola) [inline caffe2/python/examples/resnet50_trainer.py:455]: default should be True?

Comment (neobarney) [reply caffe2/python/examples/resnet50_trainer.py:455]: Default in image_input_op.cc is 0 (https://github.com/caffe2/caffe2/blob/master/caffe2/image/image_input_op.cc#L69)

So I think default in the trainer should be set to False too

**Comment by orionr** [comment]
Thank you for your contribution! PyTorch and Caffe2 are now officially merged with the PyTorch 1.0 preview release. Please submit on https://github.com/pytorch/pytorch if this is still an issue over there.

**Pull request closed without merging**